tests/test_operation.py

```diff
@@ -2724,6 +2724,51 @@ def test_mul_does_not_auto_simplify(self):
             assert qml.equal(op[0], op0 @ op1)
             assert qml.equal(op[1], op2)
 
+    class TestHamiltonianLinearCombinationAlias:
+        """Tests to validate that aliasing Hamiltonian as LinearCombination with new op math."""
+
+        def test_hamiltonian_alias_linear_combination_with_enabled_opmath(self):
+            """Test that qml.Hamiltonian is an alias for qml.ops.LinearCombination when new operator
+            arithmetic is enabled"""
+            op = qml.Hamiltonian([1.0], [qml.X(0)])
+
+            assert isinstance(op, qml.Hamiltonian)
+            assert isinstance(op, qml.ops.Hamiltonian)
+            assert isinstance(op, qml.ops.qubit.Hamiltonian)
+            assert isinstance(op, qml.ops.qubit.hamiltonian.Hamiltonian)
+            assert isinstance(op, qml.ops.LinearCombination)
+            assert op.__class__.__name__ == "LinearCombination"
+
+            qml.operation.disable_new_opmath()
+            op = qml.Hamiltonian([1.0], [qml.X(0)])
+            assert isinstance(op, qml.Hamiltonian)
+            assert not isinstance(op, qml.ops.LinearCombination)
+
+        def test_enable_opmath_multiple_times(self):
+            """Test that enabling new op math multiple times does not impact
+            Hamiltonian-LinearCombination aliasing"""
+            op = qml.Hamiltonian([1.0, 2.0], [qml.X(0), qml.Z(0)])
+            assert isinstance(op, qml.ops.LinearCombination)
+
+            qml.operation.enable_new_opmath()
+            qml.operation.enable_new_opmath()
+
+            op = qml.Hamiltonian([1.0, 2.0], [qml.X(0), qml.Z(0)])
+            assert isinstance(op, qml.ops.LinearCombination)
+
+        def test_disable_opmath_multiple_times(self):
+            """Test that disabling new op math multiple times does not impact
+            Hamiltonian-LinearCombination aliasing"""
+            op = qml.Hamiltonian([1.0, 2.0], [qml.X(0), qml.Z(0)])
+            assert isinstance(op, qml.ops.LinearCombination)
+
+            qml.operation.disable_new_opmath()
+            qml.operation.disable_new_opmath()
```

Comment:
This will fail currently, my suggestion would be to add a clause
```
def disable_new_opmath():
    """..."""
    if not active_new_opmath():
        warnings.warn(UserWarning, "Attempting to disable new opmath, but new opmath is already disabled")
        return
```

to the top of `disable_new_opmath()`

```suggestion
            with pytest.warns(UserWarning, match="Attempting to disable new opmath"):
                qml.operation.disable_new_opmath()
```

Reply:
The test passed locally for me. I'm not super keen on adding a warning for this, as it's a pretty harmless use case now that the context manager is fixed to only stop mocking if mocking is already active. I don't think we should promote users to have to check if new opmath is active before disabling/enabling like below:

```python
if qml.operation.active_new_opmath():
    qml.operation.disable_new_opmath()
...
if not qml.operation.active_new_opmath():
    qml.operation.enable_new_opmath()
```

Reply:
Yep I agree, saw now that it works to disable when its already disabled without raising an error 👍